rllib/core/rl_module/torch/tests/test_torch_rl_module.py

```diff
@@ -0,0 +1,185 @@
+            state2 = module2.get_state()
+            self.assertRaises(AssertionError, lambda: check(state, state2))
+
+            module2.set_state(state)
+            state2_after = module2.get_state()
+            check(state, state2_after)
+
+    def test_rollouts(self):
+
+        for env_name in ["CartPole-v0", "Pendulum-v1"]:
+            for fwd_fn in ["forward_exploration", "forward_inference"]:
+                for shared_encoder in [False, True]:
+                    print(
+                        f"[ENV={env_name}] | [FWD={fwd_fn}] | [SHARED={shared_encoder}]"
+                    )
+                    env = gym.make(env_name)
+
+                    if shared_encoder:
+                        config = get_shared_encoder_config(env)
+                    else:
+                        config = get_separate_encoder_config(env)
+                    module = SimplePPOModule(config)
+
+                    obs = env.reset()
+                    tstep = 0
+                    while tstep < 10:
+
+                        if fwd_fn == "forward_exploration":
+                            fwd_out = module.forward_exploration(
+                                {"obs": to_tensor(obs)[None]}
+                            )
+                            action = to_numpy(
+                                fwd_out["action_dist"].sample().squeeze(0)
+                            )
+                        elif fwd_fn == "forward_inference":
+                            # check if I sample twice, I get the same action
+                            fwd_out = module.forward_inference(
+                                {"obs": to_tensor(obs)[None]}
+                            )
+                            action = to_numpy(
+                                fwd_out["action_dist"].sample().squeeze(0)
+                            )
+                            action2 = to_numpy(
+                                fwd_out["action_dist"].sample().squeeze(0)
+                            )
+                            check(action, action2)
+
+                        obs, reward, done, info = env.step(action)
+                        print(
+                            f"obs: {obs}, action: {action}, reward: {reward}, "
+                            f"done: {done}, info: {info}"
+                        )
+                        tstep += 1
+
+    def test_forward_train(self):
+        for env_name in ["CartPole-v0", "Pendulum-v1"]:
+            for shared_encoder in [False, True]:
+                print("-" * 80)
+                print(f"[ENV={env_name}] | [SHARED={shared_encoder}]")
+                env = gym.make(env_name)
+
+                if shared_encoder:
+                    config = get_shared_encoder_config(env)
+                else:
+                    config = get_separate_encoder_config(env)
+
+                module = SimplePPOModule(config)
+                module.to("cpu")
+                module.train()
```

Comment:
Maybe add a comment here on what this `train()` call does. I know this is all PyTorch stuff and people using torch should know this, but would this state of the model have an effect on our own RLModule logics? Like would `forward_train()` still work if the module was put into `evaluate()` state?

Reply:
just showing what the training loop would look like to refer people the next developer working on trainer to this to get an idea around proper usage. `to(device)` and `.train()` are standard calls that the trainer should call around the TorchRLModule.